src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor() {
+    super();
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast instanceof Carnivore || beast.hidden) {
+      return;
+    }
+
+    beast.health -= 50;
```

Comment:
```suggestion
    if (beast instanceof Herbivore && !beast.hidden) {
          beast.health -= 50;
    }
```

Reply:
Wouldn't it be better this way?

Reply:
That line was looking off to me, but now it is more understandable :)